src/main/java/frc/robot/commands/IntakeCoral.java

```diff
@@ -0,0 +1,23 @@
+package frc.robot.commands;
+
+import com.spikes2212.command.genericsubsystem.commands.MoveGenericSubsystem;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.Storage;
+
+public class IntakeCoral extends MoveGenericSubsystem {
+
+    private static final double INTAKE_SPEED = -0.5;
+    private final Storage storage;
+
+    public IntakeCoral(Storage storage) {
+        super(storage, INTAKE_SPEED);
+        this.storage = storage;
+    }
+
+    @Override
+    public boolean isFinished() {
+        return storage.hasCoral();
```

Comment:
`MoveGenericSubsystem` implements an `isFinished` that uses the subsystem's `canMove`.
`canMove` in `MotoredGenericSubsystem` is default to return true, but you can override it how you want (ie. if the speed negative, return false if there is a coral in the system)